### ISSUE TITLE
[notifications] Split transaction bundling

### ISSUE THREAD

**Issue by alexlyp** [decred/decrediton#1718]
The current notification stream for a ticket purchase can be somewhat confusing for a user to observe.  It will show 1 success message, 1 transfer and 1 ticket purchase, in varying order.  

We should discuss ways to consume these notifications so that we always have the success message first and then we should show just 1 transaction notification for the ticket purchase.

One quick idea would be to put a "watcher" on incoming transaction notifications when the user first attempts to purchase a ticket.  Then we can do some brief verification of any normal transaction that is seen in the next N seconds.  (Does that transaction's outputs get consumed by any recent tickets?)   If that is the case then we can silence that transfer transaction notification or something.  

Open to any ideas here.

**Comment by matheusd** [comment]
One of the things I've done on the decode tx branch (nearly ready to send a separate PR for this) is to make the transaction notification handling slightly saner by bundling up a bunch of new transactions into a single state change.

Basically, I've added a delay that collects all new txs coming from the notification handler in a given amount of seconds, then sends a single new_transactions_received state change (which eventually triggers the new transaction notification).

I've also removed some code paths that triggered the notifications twice (eg: once when publishing from inside decrediton and the second when the tx notification comes in).
